### PR TITLE
NAS-113490 / 22.02 / NAS-113490: Dinstinguish csr from cert + Fix file extension when downloading csr

### PR DESCRIPTION
**Pull request by bvasilenko**
What caused the issue:
https://jira.ixsystems.com/browse/NAS-113490?focusedCommentId=144630&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-144630

Testing:  

a. download dump from the ticket, find "CSR" and "cert_type_CSR", hard-code faulty values into WS response for `certificate.query`

b. alternatively, there's a possibility to test end-to-end: create authenticator (you'll need a domain name and it's Cloudflare or AWS Route53 credentials), then create CSR, then click CSR and choose your authenticator, and then download the resulting certificate

   - AR: .crt file with size equal to 0 bytes, and error message in task manager
   - ER: .csr file with size matching the size from `ls -l` command in shell

### CODE REVIEW

**Review by RehanY147 — COMMENTED** [review]
The code looks good. Waiting for a review from @sonicaj

**Review by sonicaj — APPROVED** [review]

**Pull request merged**
